### PR TITLE
[docs] add OpExecutionContext.run

### PR DESCRIPTION
**Pull request by yuhan**
## Summary & Motivation
Feel free to push back if exposing this isn't ideal

## How I Tested These Changes
image

### CODE REVIEW

**Comment by yuhan** [comment]
* **#20957** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20957?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yuhan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by tacastillo — APPROVED** [review]
I don't have an explanation as to why I wouldn't want this. LGTM

**Review by jamiedemaria — CHANGES_REQUESTED** [review]
I agree with the doc string change, but we already have `OpExecutionContext.run` that also returns a `DagsterRun`. For `AssetExecutionContext` we deprecated `context.dagster_run` in favor of `context.run` so I would prefer to mark `OpExecutionContext.run` as public and leave `OpExecutionContext.dagster_run` non-public so it's easier to deprecate if we choose to in the future

File: python_modules/dagster/dagster/_core/execution/context/compute.py [hunk 1]
```diff
@@ -175,9 +175,10 @@ def op_config(self) -> Any:
 
     @property
     def dagster_run(self) -> DagsterRun:
-        """PipelineRun: The current pipeline run."""
+        """DagsterRun: The current run."""
```
Comment (yuhan) [inline python_modules/dagster/dagster/_core/execution/context/compute.py:178]: also updated this to be identical as `run` so we know they are the same and one is public and the other is not.

**Comment by yuhan** [comment]
thanks @jamiedemaria! updated

**Review by erinkcochran87 — APPROVED** [review]

**Review by jamiedemaria — APPROVED** [review]

**Comment by yuhan** [comment]
### Merge activity

* **Apr 2, 1:23 PM PDT**: @yuhan started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/20957).

**Pull request merged**